### PR TITLE
[XAI] Support xai basline for cls

### PR DESCRIPTION
**Pull request by dongkwan-kim01**
**Summary**
Support explain mode to MPA classification

**This PR includes**
Add ClsExplainer class to mpa/cls/explainer.py
Add explain stage

**Test Codes**
MPA standalone code will be deprecated
Can test with OTX CLI(please refer OTX cli support, [link](https://github.com/openvinotoolkit/training_extensions/pull/1311))

### CODE REVIEW

File: mpa/cls/inferrer.py [hunk 1]
```diff
@@ -100,7 +100,7 @@ def _infer(self, cfg, dump_features=False, dump_saliency_map=False):
             outputs = data_infos
         else:
             with FeatureVectorHook(model.module.backbone) if dump_features else nullcontext() as fhook:
-                with SaliencyMapHook(model.module.backbone) if dump_saliency_map else nullcontext() as shook:
+                with ActivationMapHook(model.module.backbone) if dump_saliency_map else nullcontext() as shook:
```
Comment (negvet) [inline mpa/cls/inferrer.py:103]: ASAIK `Inferrer` requires only feature vector for active sampler. `ActivationMapHook` goes to `Explainer`. Based on config, `Explainer` use `ActivationMapHook` or `EigenCamHook` to get appropriate saliency map.

**Review by negvet — CHANGES_REQUESTED** [review]

File: mpa/cls/inferrer.py (hunk 1 above)
Comment (dongkwan-kim01) [reply mpa/cls/inferrer.py:103]: Yes, then you mean delete exporting activation map from `Inferrer`?

Comment (negvet) [reply mpa/cls/inferrer.py:103]: Keep `ActivationMapHook` at the `Inferrer`, temporary. Add `ActivationMapHook` to the `Explainer` as an explain option.

> Based on config, Explainer use ActivationMapHook or EigenCamHook to get appropriate saliency map.

**Comment by dongkwan-kim01** [comment]
@wonjuleee @harimkang @sungmanc  Hello, because of Evgeny is OOO, could you please check this PR?

**Comment by dongkwan-kim01** [comment]
image
pytest passed with train & explain, for the test function in `ote_cli/utils/tests.py`, `ote_explain_testing`

File: mpa/cls/evaluator.py [hunk 2]
```diff
@@ -40,6 +40,6 @@ def run(self, model_cfg, model_ckpt, data_cfg, **kwargs):
 
         eval_cfg = cfg.get('evaluation', {})
         eval_cfg.pop('by_epoch', False)
-        results = self.dataset.evaluate(infer_results, **eval_cfg)
```
Comment (negvet) [inline mpa/cls/evaluator.py]: Please open a separate PR if needed

**Review by jaegukhyun — CHANGES_REQUESTED** [review]
Generally, It looks good to me, could you check flake8 issues once more?

**Comment by dongkwan-kim01** [comment]
@jaegukhyun Hi jaeguk, I've checked the `flake8` issue and updated it. Sorry for the late notice.

**Review by jaegukhyun — APPROVED** [review]
LGTM

**Pull request closed without merging**